### PR TITLE
Simplify hack/verify-flags-underscore.py

### LINKED ISSUE #40329: Should we move away from verify-flags.txt?
The original commit for `hack/verify-flags/known-flags.txt` claims that the purpose is for preventing flags that have `_` in the flag name. It claims that this list is used to speed up the search so that it is not obtrusive as a pre-commit hook. 

Does anyone actually use the pre-commit hooks anymore? When I joined the project, these already took so long that I almost immediately turned them off.

Can we move flag verification off of relying on known-flags.txt and no longer have to edit that file every time we add/remove a flag?

@kubernetes/test-infra-reviewers what do y'all think of this?

### PR DESCRIPTION
**Pull request by shashidharatd**
**What this PR does / why we need it**:
This PR removes the need for `hack/verify-flags/known-flags.txt` and verify-flags-underscore.py will always parse the flags from go files to check if they have underscore.

It is much faster compared to earlier checks and it does its job to check for underscore in flags.
Now:
```
# time ./hack/verify-flags-underscore.py 
real	0m1.638s
user	0m1.560s
sys	0m0.076s
```
Before:
```
# time ./hack/verify-flags-underscore.py 
real	0m22.585s
user	0m22.464s
sys	0m0.112s
```

It has become a pain to keep adding new flag to `known-flags.txt` whenever a new flag is introduced. with this PR this is step is not required anymore.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #40329  #50319

**Special notes for your reviewer**:

**Release note**:
```
NONE
```
/cc @fejta @mtaufen

### CODE REVIEW

**Comment by ixdy** [comment]
wow, nice PR!

have you tried explicitly adding a goflag with underscores to make sure it still works (fails) as expected?

/retest

**Comment by shashidharatd** [comment]
yes @ixdy, i have checked that and it works.

File: hack/verify-flags-underscore.py [hunk 1]
```diff
@@ -184,88 +116,26 @@ def get_flags(rootdir, files):
...
         files = get_all_files(rootdir)
-    files = normalize_files(rootdir, files)
-
-    check_known_flags(rootdir)
-
-    flags = get_flags(rootdir, files)
-    flagRE = flags_to_re(flags)
-
-    bad_lines = []
-    # walk all the file looking for any flag that was declared and now has an _
```
Comment (ixdy) [inline hack/verify-flags-underscore.py]: I'm not sure why we had this check (I'm guessing it's what caused the slowness), so it seems fine to me to remove this.

File: hack/verify-flags-underscore.py [hunk 2]
```diff
@@ -184,88 +116,26 @@ def get_flags(rootdir, files):
                 new_excluded_flags.add(flag)
             if not "-" in flag:
                 continue
-            if flag not in flags:
-                new_flags.add(flag)
+            flags.add(flag)
```
Comment (ixdy) [inline hack/verify-flags-underscore.py]: we collect this set of flags to return, but we don't actually use the return value anymore. I don't think collecting this set is even necessary now.

Comment (shashidharatd) [inline hack/verify-flags-underscore.py]: Agree, thats not needed. we should just remove it.

**Comment by shashidharatd** [comment]
Thanks @ixdy

**Comment by shashidharatd** [comment]
@ixdy, I had to rebase to resolve conflict. Can you lgtm again. Thanks !

**Comment by luxas** [comment]
Thanks for doing this @shashidharatd !!

**Pull request merged**